regression-tests/run-tests.sh

```diff
@@ -130,31 +130,39 @@ if [[ "$cxx_compiler" == *"cl.exe"* ]]; then
     compiler_version=$(cl.exe)
 else
     compiler_cmd="$cxx_compiler -I../../../include -std=c++20 -pthread -o "
-    
-    compiler_ver=$("$cxx_compiler" --version)
-    if [[ "$compiler_ver" == *"Apple clang version 14.0"* ]]; then
+    compiler_version=$("$cxx_compiler" --version)
+
+    # We don't currently support Apple Clang 15 so try and switch to 14
+    if [[ "$compiler_version" == *"Apple clang version 15.0"* ]]; then
```

Comment:
[Question] Are you sure Apple Clang 15 was selected? The [GitHub runner](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md) states that Clang 15 s installed from Homebrew. I checked and the output of `$(brew --prefix llvm@15)/bin/clang` is:
```
Homebrew clang version 15.0.7
Target: x86_64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm@15/bin
```

Reply:
No, I'm not 100% sure, this is very much some investigative work!

All I can say for certain right now is that on some macOS job runs (e.g. the one linked above, and I've seen it other times too), `clang++` does not resolve to Apple Clang 14. That's why the script produces the `not found for compiler: 'clang++'` error message.

One of the updates in this PR is to provide more info about why the job fails. If this PR is merged then in the future we'll get more info due to this:
```
printf "Unhandled compiler version:\n$compiler_version\n\n"
```
That may show `Apple clang 15.0 etc` or something else.

Yeah, the `macos-13` runner confusingly has both LLVM Clang installed as well as multiple versions of Xcode (with Apple Clang).

![image](https://github.com/hsutter/cppfront/assets/2060747/07252642-5455-4382-9bb4-205cea0d5e1c)

Xcode 14.3 is meant to be the default so `clang++` would resolve to Apple Clang 14, but I suspect there's a problem which is causing Xcode 15 to be active. But, it could also somehow be invoking LLVM Clang 15. To me that seemed less likely because, like you noted, it's installed via Homebrew and needs the `brew --prefix llvm@15` to get the correct path.